src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,155 @@
+import "@polymer/paper-icon-button/paper-icon-button";
+import "@polymer/paper-input/paper-input";
+import "@polymer/paper-item/paper-icon-item";
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-listbox/paper-listbox";
+import memoizeOne from "memoize-one";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import { compare } from "../../common/string/compare";
+import { LocalizeFunc } from "../../common/translations/localize";
+import computeStateName from "../../common/entity/compute_state_name";
+
+export interface DeviceTrigger {
+  platform: string;
+  domain: string;
+  device_id: string;
+  entity_id?: string;
+  type: string;
+}
+
+const fetchDeviceTriggers = (hass, deviceId) =>
+  hass.callWS<DeviceTrigger[]>({
+    type: "device_automation/list_triggers",
+    device_id: deviceId,
+  });
+
+function isObject(v) {
+  return "[object Object]" === Object.prototype.toString.call(v);
+}
+
+JSON.sort = function(o) {
+  if (Array.isArray(o)) {
+    return o.sort().map(JSON.sort);
+  } else if (isObject(o)) {
+    return Object.keys(o)
+      .sort()
+      .reduce(function(a, k) {
+        a[k] = JSON.sort(o[k]);
+
+        return a;
+      }, {});
+  }
+
+  return o;
+};
+
+class HaEntityPicker extends LitElement {
+  public hass?: HomeAssistant;
+  public localize?: LocalizeFunc;
+  @property() public label?: string;
+  @property() public value?: string;
+  @property() public deviceId?: string;
+  @property() public triggers?: DeviceTrigger[];
+
+  private _sortedTriggers = memoizeOne((triggers?: DeviceTrigger[]) => {
+    return triggers || [];
+  });
+
+  protected render(): TemplateResult | void {
+    return html`
+      <paper-dropdown-menu-light .label=${this.label}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._value}
+          attr-for-selected="data-trigger"
+          @iron-select=${this._triggerChanged}
+        >
+          <paper-item
+            data-trigger=${JSON.stringify({
```

Comment:
Don't do this. Instead set it as a property. `.trigger=${{device_id: "", platform: ""}`

Reply:
Fixed.